### PR TITLE
Take Nexus context into account in REST calls

### PR DESCRIPTION
**Pull request by manics**
`urljoin(self.config.url, '/service/rest/')` discards the path component of `self.config.url` since the leading `/` on the second argument indicates it's an absolute URL: https://stackoverflow.com/questions/10893374/python-confusions-with-urljoin

This should fix it. Includes mock tests.

### CODE REVIEW

File: src/nexuscli/nexus_client.py [hunk 1]
```diff
@@ -117,7 +117,10 @@ def rest_url(self):
 
         :rtype: str
         """
-        url = urljoin(self.config.url, '/service/rest/')
+        baseurl = self.config.url
```
Comment (bt-thiago) [inline src/nexuscli/nexus_client.py]: Wouldn't it be better to enforce this in the NexusConfig class?  i.e.: https://github.com/thiagofigueiro/nexus3-cli/blob/master/src/nexuscli/nexus_config.py#L45

File: tests/test_nexus_client.py [hunk 2]
```diff
@@ -56,3 +57,32 @@ def test_split_component_path_errors(
...
+            self.status_code = 200
+
+        def json(self):
+            return '{}'
+
+    mocker.patch('requests.request', return_value=MockResponse())
+
+    NexusClient(NexusConfig(url=url))
+    requests.request.assert_called_once_with(
+        auth=('admin', 'admin123'), method='get', stream=True,
```
Comment (bt-thiago) [inline tests/test_nexus_client.py]: the default credentials are  in nexus_config.DEFAULTS: https://github.com/thiagofigueiro/nexus3-cli/blob/master/src/nexuscli/nexus_config.py#L9-L10
```suggestion
        auth=(nexus_config.DEFAULTS['username'], nexus_config.DEFAULTS['password']), method='get', stream=True,
```

**Comment by bt-thiago** [comment]
Thank you for your contribution.

File: tests/test_nexus_client.py [hunk 3]
```diff
@@ -1,9 +1,11 @@
 # -*- coding: utf-8 -*-
 import pytest
+import requests
 
 import nexuscli
 from nexuscli import exception
-from nexuscli.nexus_client import NexusClient
+from nexuscli.nexus_client import NexusClient, NexusConfig
```
Comment (bt-thiago) [inline tests/test_nexus_client.py]: `NexusConfig` is in `nexus_config` IIRC.

Comment (manics) [reply tests/test_nexus_client.py]: You're right, I've fixed it

**Comment by manics** [comment]
@bt-thiago Are you happy with this?

**Pull request merged**